docs/src/apalache/tuning.md

```diff
@@ -91,6 +86,47 @@ steps.
 You can also use this option to check different parts of an invariant on
 different machines to speed up turnaround time.
 
+## Z3 Tuning Parameters
+
+Z3 has a very large number of [Z3 Parameters][]. If you have a CLI version of Z3
+installed, you can see a complete list of supported parameters by running `z3`:
+
+```sh
+z3 -pd
+```
+
+In Apalache, you can pass a Z3 parameter `x.y.z=v` as a fine-tuning parameter:
+
+```
+z3.x.y.z=v
+```
+
+For example, to change the SAT and SMT restarts strategies to static:
+
+```
+z3.sat.restart = static
+z3.smt.restart_strategy = 3
+```
+
+Sometimes, the above settings help Apalache to show unsatisfiability faster.
+
+You can also employ Z3 parallelization by setting the number of threads:
+
+```
+z3.sat.threads=10
+```
+
+Technically, Apalache propagates all the parameters that start with `z3.` to Z3.
+However, some of these parameters fail in the solver, even when they work in the
+command line. Hence, you have to experiment with the choice of parameters.
+
+## Randomization
+
+**DEPRECATED:** You can now pass the seeds directly to Z3 via [Z3 Tuning Parameters](#z3-tuning-parameters).
+
+`smt.randomSeed=<int>` passes the random seed to `z3` (via `z3`'s parameters
+`sat.random_seed` and `smt.random_seed`).
```

Comment:
`smt.randomSeed` propagates to more than just `sat.random_seed` and `smt.random_seed`:

https://github.com/apalache-mc/apalache/blob/8630a60cb48b3c6f1d88c8e43f470182edae083d/tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala#L1029-L1035

I don't think we should deprecate this.

Reply:
Yep, I was not sure about this either. The only issue is that some of the above settings conflict with more restricted logics